format/nexus.py

```diff
@@ -223,12 +232,12 @@ def visit_dependencies(nx_file, item, visitor=None):
     """
     Walk the dependency chain and call a visitor function
     """
-    dependency_chain = []
+    dependency_chain = set()
```

Comment:
Initial comment: I am slightly worried here about the mix of type changes which are fairly obvious with changing the dependency hierarchy discovery logic to use `set` not `list` - not asking for any changes, just suggesting for future work we separate logical and syntax changes

Reply:
sorry, that just seemed an obvious O(n²)->O(n) campsite improvement.
`dependency_chain` is only ever used to append elements to, and when an element already existed in it an Exception is raised. After that it is discarded either way.

Reply:
No need for sorry, and makes sense - just flagged as non-syntax 🙂